src/WebJobs.Extensions.DurableTask/Bindings/OrchestrationTriggerAttributeBindingProvider.cs

```diff
@@ -197,11 +215,14 @@ public Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
                 return Task.FromResult<IListener>(listener);
             }
 
-            private static string OrchestrationContextToString(DurableOrchestrationContext arg)
+            private string OrchestrationContextToString(DurableOrchestrationContext arg)
             {
-                var history = JArray.FromObject(arg.History, DefaultSerializer);
                 var input = arg.GetInputAsJson();
 
+                var usesExternalPowerShellSDK = this.platormInformation.UsesExternalPowerShellSDK();
+                JsonSerializer serializer = usesExternalPowerShellSDK ? TypePreservingSerializer : DefaultSerializer;
```

Comment:
This change does slightly concern me with the longevity of it. The `$type` is Newtonsoft.Json specific (I think it has its own way of writing / reading the type name) - so how robust is this for if we want to change to STJ later on? Even if the change is only on the other end of this (whoever needs the type information).

Also, will this affect any other serialization with powershell? Will admin APIs be affected?

Reply:
Good point on $type being Newtonsoft.JSON specific. I think we don't have to be too concerned about longetivity of this strategy since the JSON-based OOProc protocol implemented in this path should be "soon" deprecated in favor of the protobuf-based protocol that C#-isolated and Java utilize today.

Currently, I plan on GA'ing the new PS SDK with the protobuf implementation (not supported today) and so, as soon as that's enabled, we should be able to deprecate this JSON-based type-preserving serializer.

As for whether this affects any other serialization with PowerShell. I don't think so. This is only serializing the DF payload. All other data going into the PS worker is serialized independently.

So, to summarize - I think of this change as a temporary means of enabling the new PS SDK to be previewed by customers. Then I'll implement the protobuf protocol prior to GA, so we should't have to support this change in the long run. 

Does that make sense and/or seem reasonable?

Reply:
Ah great, that plan sounds good to me!